### PR TITLE
[Reviewer=BJR2][BGCF-67]  Add Replace Function to Python RE2 Bindings

### PR DESCRIPTION
**Pull request by jack-atack**
This has been modelled off the existing _do_search bindings which call the RE2 match function.  

I've called the function replace rather than global_replace (which is the name of the RE2 function), as I think replace is more intuitive (and I am not aware of plans to have both singular and global replace used separately)

### CODE REVIEW

File: _re2.cc [hunk 1]
```diff
@@ -27,7 +27,7 @@
  */
 #define PY_SSIZE_T_CLEAN
 #include <Python.h>
-
+#include <iostream>
```
Comment (huws) [inline _re2.cc]: Is this really required?  If not, remove it.

File: _re2.cc [hunk 2]
```diff
@@ -111,6 +112,9 @@ static PyMethodDef regexp_methods[] = {
     "match(string[, pos[, endpos]]) --> match object or None.\n"
     "    Matches zero or more characters at the beginning of the string"
   },
+  {"replace", (PyCFunction)regexp_replace, METH_VARARGS | METH_KEYWORDS,
+    ""
```
Comment (huws) [inline _re2.cc]: Fill in the function comment, c.f. the other examples.

**Review by huws — COMMENTED** [review]
Looks good.  Two very minor comments inline.  I think I would have more directly mirrored the underlying method names, but I don't really care.  This is trivial to change in future, if that's what we want.

File: _re2.cc (hunk 1 above)
Comment (jack-atack) [reply _re2.cc]: Good spot, that's left in from my debugging

**Pull request merged**